Passing dot as current working directory fails

Passing `.` to the `--cwd` option fails. As a workaround, I pass `$PWD` instead. I guess the given path should be resolved before entering any temporary directory? Didn't look at the code yet.

<details><summary>Full traceback (click to expand)</summary>

```python
% pipx run terminhtml --cwd . "make format" > output.html
Traceback (most recent call last):

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 113, in _get_real_output_and_context_from_output_lines
    command_marker_indices = CommandMarkerIndices.find(lines)

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 204, in find
    raise CannotFindPersistenceLineMarkersException(

terminhtml._exc.CannotFindPersistenceLineMarkersException: Could not find all command markers in lines: [LineOutput(line="make: *** No rule to make target 'format'.  Stop.", time=datetime.datetime(2022, 9, 25, 15, 57, 46, 216178), line_ending=<LineEnding.CRLF: '\r\n'>, prompt_output=None), LineOutput(line='', time=datetime.datetime(2022, 9, 25, 15, 57, 46, 216390), line_ending=<LineEnding.CRLF: '\r\n'>, prompt_output=None)]. Markers were: markers=CommandMarkerIndices(begin_persistence=-1, end_persistence=-1, begin_path=-1, end_path=-1, begin_env=-1, end_env=-1)


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 34, in run
    return _run(

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 333, in _run
    real_output, new_context = _get_real_output_and_context_from_output_lines(

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 117, in _get_real_output_and_context_from_output_lines
    raise CommandInternalException(

terminhtml._exc.CommandInternalException: Command had non-zero exit code


The above exception was the direct cause of the following exception:


Traceback (most recent call last):

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/bin/terminhtml", line 8, in <module>
    sys.exit(main())

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/cli.py", line 157, in main
    typer.run(run_commands_create_html)

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/cli.py", line 138, in run_commands_create_html
    term = TerminHTML.from_commands(

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/main.py", line 59, in from_commands
    command_results = _run_commands_create_command_results(

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/main.py", line 137, in _run_commands_create_command_results
    return run_commands(

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 96, in run_commands
    out_command = run(command, last_context, input=this_command_input)

  File "/home/pawamoy/.local/pipx/.cache/78b42e3944fa77f/lib/python3.10/site-packages/terminhtml/runner/main.py", line 56, in run
    raise UserCommandException(

terminhtml.exc.UserCommandException: Command failed: make format due to Command had non-zero exit code as part of running commands=['make format'] setup_command='' input=None
context=cwd=PosixPath('.') env=None
output:
make: *** No rule to make target 'format'.  Stop.
```

</details>

--- a/terminhtml/cli.py
+++ b/terminhtml/cli.py
@@ -51,7 +51,7 @@ def run_commands_create_html(
     cwd: Optional[Path] = typer.Option(
         None,
         "--cwd",
-        "-c",
+        "-d",
         help="Working directory to run commands in. "
         "If not passed, defaults to a temporary directory.",
     ),

--- a/terminhtml/runner/main.py
+++ b/terminhtml/runner/main.py
@@ -76,10 +76,11 @@ def run_commands(
 
     input = input or []
     orig_dir = os.getcwd()
+    absolute_cwd = cwd.absolute() if cwd else None
     with tempfile.TemporaryDirectory() as tmpdir:
         os.chdir(tmpdir)
-        if cwd:
-            begin_cwd = cwd
+        if absolute_cwd:
+            begin_cwd = absolute_cwd
         else:
             begin_cwd = Path(tmpdir)
         last_context = RunnerContext(cwd=begin_cwd)

--- a/tests/dirutils.py
+++ b/tests/dirutils.py
@@ -1,0 +1,11 @@
+import os
+from contextlib import contextmanager
+from pathlib import Path
+
+
+@contextmanager
+def change_directory_to(path: Path):
+    current_path = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(current_path)

--- a/tests/e2e/test_cwd.py
+++ b/tests/e2e/test_cwd.py
@@ -1,0 +1,30 @@
+import pytest
+
+from tests.config import INPUT_FILES_DIR
+from tests.dirutils import change_directory_to
+from tests.e2e.cli_stub import run_cli
+
+
+@pytest.mark.parametrize(
+    "flag",
+    ["-d", "--cwd"],
+)
+def test_command_runs_in_absolute_cwd_when_passed(flag: str):
+    cwd = INPUT_FILES_DIR
+    result = run_cli(f"{flag} {cwd.absolute()} 'ls -l'")
+    text = result.output
+    assert result.exit_code == 0
+    assert "rich_progress_bar.html" in text
+
+
+@pytest.mark.parametrize(
+    "flag",
+    ["-d", "--cwd"],
+)
+def test_command_runs_in_current_directory_when_passed_as_dot(flag: str):
+    cwd = INPUT_FILES_DIR
+    with change_directory_to(cwd):
+        result = run_cli(f"{flag} . 'ls -l'")
+        text = result.output
+        assert result.exit_code == 0
+        assert "rich_progress_bar.html" in text

